docs/unit-test-framework-tutorial.md

```diff
@@ -0,0 +1,184 @@
+
+  - The `TEST(<test_name>)` essentially gets replaced (by the preprocessor) with a test function called <test_name>, where <test_name> is any valid C++ function name and <test_name> will be the name of the new test function. You do **NOT** need to put quotes around <test_name>, and if you do you’ll get a compiler error.
+
+  - `TEST_MAIN()` gets replaced by a `main()` function that detects and runs all of the test cases you defined using the `TEST()` macro. Unlike in P1 where you had to explicitly call your test functions from `main()`, this framework handles that for you!
+
+  - `ASSERT_TRUE()` is one of several special test assertion preprocessor macros that you can use to check conditions in your test cases. You’ll be using these instead of `assert()` in your unit tests. These will be demonstrated in more detail in the next section.
+
+Compile and run this test case with the following two commands:
+
+```
+$ g++ -Wall -Werror -pedantic -O1 -std=c++11 lab02_tests.cpp unit_test_framework.cpp -o lab02_tests.exe
+
+$ ./lab02_tests.exe
+```
+
+Output:
+```
+Running test: numbers_are_equal
+PASS
+Running test: true_is_true
+PASS
+
+*** Results ***
+** Test case 'numbers_are_equal': PASS
+** Test case 'true_is_true': PASS
+*** Summary ***
+Out of 2 tests run:
+0 failure(s), 0 error(s)
+```
+
+Another nice feature of the framework is that we can tell it to run only a subset of our test cases. If we wanted to only run the test `numbers_are_equal`, we could do it with this command:
+```
+$ ./lab02_tests.exe numbers_are_equal
+```
+
+Output:
+```
+Running test: numbers_are_equal
+PASS
+
+*** Results ***
+** Test case 'numbers_are_equal': PASS
+*** Summary ***
+Out of 1 tests run:
+0 failure(s), 0 error(s)
+```
+
+You can pass any number of test names as command line arguments, and it will only run the ones you’ve specified.
+
+## Special Test Assertions
+One of the main reasons for using the special assertions provided by the framework is that they allow the framework to run all of your tests and report which ones passed and which ones failed. As you may have noticed, when you use regular assert() in your test cases, they automatically stop at the first failure. This can make it difficult to debug errors in one test that are actually caused by function whose test cases didn’t get a chance to run yet.
+
+Here is a summary of all the special assertions that the framework provides:
+
+Assertion | Description
+------------ | -------------
+<code>ASSERT_EQUAL(<i>first</i>, <i>second</i>)</code> | If first == second evaluates to false, the test will fail. **Note:** Do not use this if first and second are not comparable using the == operator. Other than this restriction, first and second may be any type.

```

Comment:
Use backtic instead of HTML's `<code>`

Reply:
I did that in order to be able to have the italics within.  I don't believe it is possible with backticks.